File: 985-sum-of-even-numbers-after-queries/985-sum-of-even-numbers-after-queries.py
class Solution:
    def sumEvenAfterQueries(self, nums: List[int], queries: List[List[int]]) -> List[int]:
        sums = 0
        l=[]
        for i in nums:
            if i%2==0:
                sums+=i
        for i in queries:
            if nums[i[1]]%2==0:
                sums-=nums[i[1]]
            nums[i[1]]+=i[0]
            if nums[i[1]]%2==0:
                sums+=nums[i[1]]
            l.append(sums)
        return l
    

File: 985-sum-of-even-numbers-after-queries/985-sum-of-even-numbers-after-queries.java
class Solution {
    public int[] sumEvenAfterQueries(int[] nums, int[][] queries) {
        int s = 0;
        int [] ans = new int[queries.length];
        for(int i:nums){
            if(i%2==0) s+=i;
        }
        int i=0;
        for(int [] a : queries){
            if (nums[a[1]]%2==0) s-=nums[a[1]];
            nums[a[1]]+=a[0];
            if (nums[a[1]]%2==0) s+=nums[a[1]];
            ans[i++]=s;
        }
        return ans;
    }
}